File: src/main/java/de/litexo/model/external/ExplorerDirectory.java
package de.litexo.model.external;

import lombok.Data;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.List;

@Data
@Accessors(chain = true)
public class ExplorerDirectory {
    String id;
    String name;
    String absolutePath;
    List<ExplorerFile> files = new ArrayList<>();
}

File: ui/src/app/api/models/explorer-directory.ts
/* tslint:disable */
/* eslint-disable */
import { ExplorerFile } from './explorer-file';
export interface ExplorerDirectory {
  absolutePath?: string;
  dirs?: Array<ExplorerDirectory>;
  files?: Array<ExplorerFile>;
  id?: string;
  name?: string;
  relativePath?: string;
}
